pkg/csvcopy/batch_error.go

```diff
@@ -0,0 +1,62 @@
+package csvcopy
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/timescale/timescaledb-parallel-copy/pkg/batch"
+)
+
+// BatchHandlerSaveToFile saves the errors to the given directory using the batch start row as file name.
+func BatchHandlerSaveToFile(dir string, next BatchErrorHandler) BatchErrorHandler {
+	return BatchErrorHandler(func(batch batch.Batch, reason error) error {
+		err := os.MkdirAll(dir, os.ModePerm)
+		if err != nil {
+			return fmt.Errorf("failed to ensure directory exists: %w", err)
+		}
+
+		fileName := fmt.Sprintf("%d.csv", batch.Location.StartRow)
```

Comment:
why not add some wording to the filename as well like `row_offset_%d.csv` it might make it easier to understand the output